docs/reference/commandline/login.md

```diff
@@ -162,19 +157,93 @@ the credentials from the default store.
 Credential helpers are specified in a similar way to `credsStore`, but
 allow for multiple helpers to be configured at a time. Keys specify the
 registry domain, and values specify the suffix of the program to use
-(i.e. everything after `docker-credential-`).
-For example:
+(i.e. everything after `docker-credential-`). For example:
 
 ```json
 {
   "credHelpers": {
-    "registry.example.com": "registryhelper",
-    "awesomereg.example.org": "hip-star",
-    "unicorn.example.io": "vcbait"
+    "myregistry.example.com": "secretservice",
+    "docker.internal.example": "pass",
   }
 }
 ```
 
+## Examples
+
+### Authenticate to Docker Hub using device code
+
+By default, the `docker login` command authenticates to Docker Hub, using
+a device code flow. This flow lets you authenticate to Docker Hub without
+entering your password. Instead, you visit a URL in your browser, enter a
+code, and authenticate.
+
+```console
+$ docker login
+
+USING WEB BASED LOGIN
+To sign in with credentials on the command line, use 'docker login -u <username>'
+
+Your one-time device confirmation code is: LNFR-PGCJ
+Press ENTER to open your browser or submit your device code here: https://login.docker.com/activate
+
+Waiting for authentication in the browser…
+```
+
+After entering the code in your browser, you are authenticated to Docker Hub
+using the account you're currently signed in with on the Docker Hub website or
+in Docker Desktop. If you aren't signed in, you are prompted to sign in after
+entering the device code.
+
+### Authenticate to a self-hosted registry
+
+If you want to authenticate to a self-hosted registry you can specify this by
+adding the server name.
+
+```console
+$ docker login registry.example.com
+```
+
+By default, the `docker login` command assumes that the registry listens on
+port `5000`. If the registry listens on a different port, you can specify it
+by adding the port number to the server name.
```

Comment:
I don't think it assumes port 5000, but will default to `443` (TLS) / `80` (non-TLS). (The examples for the registry tend to use port 5000 though, so maybe that's what you assumed here?)

Reply:
ops yea my bad!